core/app/[locale]/(default)/[...rest]/page.tsx

```diff
@@ -1,42 +1,27 @@
-import { Page as MakeswiftPage } from '@makeswift/runtime/next';
-import { getSiteVersion } from '@makeswift/runtime/next/server';
-import { notFound } from 'next/navigation';
+import { locales } from '~/i18n/routing';
+import { client, Page } from '~/lib/makeswift';
 
-import { defaultLocale, locales } from '~/i18n/routing';
-import { client } from '~/lib/makeswift/client';
+interface PageParams {
+  locale: string;
+  rest: string[];
+}
 
-export async function generateStaticParams() {
+export async function generateStaticParams(): Promise<PageParams[]> {
   const pages = await client.getPages().toArray();
 
-  return pages
-    .filter((page) => page.path !== '/')
-    .flatMap((page) =>
-      locales.map((locale) => ({
-        rest: page.path.split('/').filter((segment) => segment !== ''),
-        locale: locale === defaultLocale ? undefined : locale,
-      })),
-    );
-}
-
-interface Props {
-  params: Promise<{
-    locale: string;
-    rest: string[];
-  }>;
+  return pages.filter((page) => page.path !== '/').flatMap((page) => localesFanOut(page.path));
 }
 
-export default async function CatchAllPage({ params }: Props) {
+export default async function CatchAllPage({ params }: { params: Promise<PageParams> }) {
   const { rest, locale } = await params;
   const path = `/${rest.join('/')}`;
 
-  const snapshot = await client.getPageSnapshot(path, {
-    siteVersion: await getSiteVersion(),
-    locale: locale === defaultLocale ? undefined : locale,
-  });
-
-  if (snapshot == null) return notFound();
-
-  return <MakeswiftPage snapshot={snapshot} />;
+  return <Page locale={locale} path={path} />;
 }
 
-export const runtime = 'nodejs';
+function localesFanOut(path: string): PageParams[] {
+  return locales.map((locale) => ({
+    rest: path.split('/').filter((segment) => segment !== ''),
+    locale,
+  }));
+}
```

Comment:
nit: I think extracting out this function might be a bit harder for users of Catalyst to follow than leaving it inline in `generateStaticParams`. The reason is because the return type is coupled to the route params and so the function isn't reusable in other contexts. Also, I think it's helpful for `generateStaticParams` to be fully contained to remove the need to jump around to figure out what's being statically generated

Reply:
If the function was incomprehensible because of the chained `map` and `flatMap` perhaps separating it out into different variables might help with readability. I just think that there's a cost to having to go find the function being called to understand how the path is being mapped to the route params (in particular because it doesn't take `locales` as a param and reads it from module context)

Reply:
> I just think that there's a cost to having to go find the function being called to understand how the path is being mapped

Sure, there is a cost to everything; inlining the logic or putting it into a variable makes you pay the cost of parsing and understanding code with multiple variables coming from different scopes upfront. Giving a piece of logic a mnemonic name and tacking it nearby gives you a chance to understand the bigger picture at glance and make a decision whether you want to explore the details deeper now or later (or never). Breaking down the problem into smaller semantic steps with lesser parts is worth a lot more to me than the cost of clicking on a function to jump to its definition when I'm ready to see the details.

> in particular because it doesn't take locales as a param and reads it from module context

The global `locales` being buried inside of an arrow function inside of `flatMap` is worse, isn't it?